src/transaction/client_credentials.cpp

```diff
@@ -116,7 +116,7 @@ clientids::set_program_name (const char *program_name_arg)
 {
   char prgname_buf[PATH_MAX] = {'\0'};
 
-  if (program_name_arg == NULL || basename_r (program_name_arg, prgname_buf, PATH_MAX) < 0)
+  if (program_name_arg == NULL)
```

Comment:
Is this intended?

Reply:
Isn't the return value of basename_r always positive or 0.

Reply:
The return value would seems to be always positive. The function returns an int though, so it could be modified to return negative values).

Removing basename_r call has other side effects too. Is it intended to remove the calculation of prgname_buf?

And last question, what is the link between basename_r call removal and the current task?